### DISCUSSION TITLE
Dexcom G7 very often able to pair, but unable to get readings

### DISCUSSION THREAD

**Discussion by charludo** [NightscoutFoundation/xDrip#2953] [General]
Greetings,

for the past couple of months I have had very bad luck using xdrip with the G7. 2 out of 3 times, the sensor will pair (sensor status: warming up, last connected: x minutes ago, and so on), but then once the 30 minute warmup is done, xdrip will not be able to get any readings from the sensor whatsoever (tried multiple phone restarts, location services are enabled, waited multiple hours).

I was leaning towards blaming Dexcom and their shoddy quality control (and they have been sending replacements without complaints), however I just noticed with another new sensor which exhibited this behavior that upon removing it from my arm, the status in xdrip quickly updated to "sensor error".

So I'm opening this discussion, mainly to see if others are experiencing similar symptoms or failure rates.

**Comment by Navid200** [comment]
Hello,

Would you please make sure to have the latest Nightly installed?  https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.16

Then, if any settings are different, please correct them: https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

Then, if the problem persists, please enable OB1 debugging: https://navid200.github.io/xDrip/docs/Dexcom/OB1Debug.html

Wait for the problem to occur again.
Then, go to logs and upload adding a note to identify.

Then, post here and add the identifying content so that association can be made between the logs and this thread.

**Reply by charludo** [reply]
Thanks, I'm on the current nightly (though the issue has been present in all versions I've tested back through at least may).

Settings are exactly as described.

Didn't know about the debugging - I'll enable it and update here once the issue appears again!

**Reply by charludo** [reply]
@Navid200 sorry to complicate this further, however I noticed yesterday that I was wrong I overlooked that one setting differed, namely I had "Avoid Scanning" enabled.

I don't know if that's connected to my issue (haven't changed a sensor yet), however I did notice that after disabling "Avoid Scanning", I'm no longer able to get measurements at all unless GPS is enabled.

Re-activating "Avoid Scanning" has resolved this (thankfully, since at least on my phone GPS is a large battery drain).

Is this intentional / documented somewhere?

**Reply by Navid200** [reply]
Android Location service must be enabled on Android.  Otherwise, xDrip will not be able to connect.
I don't believe this is an xDrip limitation.  I don't think Dexcom app can operate either if you disable the location service.

**Reply by charludo** [reply]
For the initial connection, yes. (AFAIK this is to ensure that user and Bluetooth device are in roughly the same location.) 

After that, GPS does not have to remain active in the official Dexcom app, and neither does it in xdrip if "Avoid Scanning" is enabled. (This includes re-opening the connection to the sensor after prolonged signal loss.) 

I guess my issue could possibly be connected to this phenomenon. I'll see in the coming weeks.

But regardless, I guess it could be of some interest to others as well who don't want to keep GPS enabled constantly.

**Comment by tzachi-dar** [comment]
Bluetooth and location permissions are related because if an app has the ability to scan for bluetooth devices it can get the location.
The reason is that there are bluetooth beacons that people put and allows the application to know where it is.
See https://www.wiliot.com/bluetooth-location-beacon-for-asset-tracking

**Reply by charludo** [reply]
Yep, I'm not debating that. Just pointing out that with "Avoid Scanning" enabled, location services are not required beyond the initial pairing with the sensor.

**Reply by Navid200** [reply]
What is the problem really?
Is xDrip working for you if you follow our recommended settings?
We have asked you to disable avoid scanning.  Not just until it pairs.  We have asked you to leave it disabled.

Have you done that?

**Reply by Navid200** [reply]
And also, the location access requirement is explained here:
https://navid200.github.io/xDrip/docs/Location.html

**Reply by charludo** [reply]
The issue was that very often (two thirds of the time) I can pair, but then do not get measurements after the warmup period. (This happens irregardless of location services being on.)

This might just be faulty sensors, but 66% failure rate seems very high, so I wanted to see if other people have similar issues (hence this discussion thread). 

After your initial comment I went back and checked again if all settings were as you recommended. "Avoid Scanning" was enabled against those recommendations. 

I disabled this, then noticed that my (already up and running previously) sensor stopped working with xdrip unless I kept GPS enabled at all times. Re-enabling "Avoid Scanning" fixes this, and measurements are once again read from the sensor when location services are disabled. 

So the questions remaining are:

- is it intentional that without "Avoid Scanning" enabled, GPS has to be switched on at all times, not just for pairing (which is what the Android specs require; you don't, for example, need to keep GPS on to use your Bluetooth headsets after initial pairing either) 
- is there a reason why we should not enable "Avoid Scanning"? Was it already known / documented somewhere that this option alleviates the need to keep GPS on constantly? 
- is "Avoid Scanning" responsible for my initial issue (pairing works, readings don't)? 

That last one I'll try to figure out next time I need to start a new sensor (sometime next week) by leaving "Avoid Scanning" enabled (as I originally had it), hopefully encountering the same issue again, and seeing if disabling it fixes the issue (unless you have a better idea how to figure this out?).

The former two questions, it would be great if you or someone else involved with the project has an answer to! 😄


>We have asked you to disable avoid scanning. Not just until it pairs. We have asked you to leave it disabled.

>Have you done that?

No, simply because I can't activate a new sensor for another 8 days. But for now, I'll leave it enabled, because keeping GPS on 24/7 is not an option for me.

If disabling "Avoid Scanning" fixes my original issue, I'll disable it -> start new sensor, wait until first measurements arrive -> enable it again.

**Reply by charludo** [reply]
> And also, the location access requirement is explained here:
> https://navid200.github.io/xDrip/docs/Location.html

Yes, this permission is granted and I have not revoked it at any point.

I really am talking about GPS.

**Reply by Navid200** [reply]
There is no need for GPS.
But, the location service must be enabled.

**Reply by charludo** [reply]
~~Then it looks like I'm having a legitimate issue here.~~

- ~~With location services enabled, GPS enabled, Avoid Scanning disabled: I get measurements.~~
- ~~With location services enabled, GPS disabled, Avoid Scanning disabled: I do not get any measurements.~~
- ~~With location services enabled, GPS enabled or disabled, Avoid Scanning enabled: I get measurements.~~

I'm sorry, I misunderstood what was meant by location services (my phone is not set to English).

But what I was saying previously remains true: if Avoid Scanning is on and location services are off, xDrip will still keep receiving measurements from the sensor. 

So I'm wondering, why is it recommended to switch Avoid Scanning off.

**Comment by charludo** [comment]
So it turns out this was probably really a QA issue. Soon after opening this discussion sensors for Q3 arrived from Dexcom, I started using those and haven't had an issue since.

Note that I kept "Avoid Scanning" enabled permanently, and Location Services disabled except for when pairing with a new sensor. I have experienced no issues related to this.
